RAC-190: Documentation of DSM (#12828)
akeneo/pim-community-dev: akeneo-design-system/CONTRIBUTING.md

> # Contribute

Reviewer: ```suggestion
# Contributing
```

> We prefer consistency above all. So that the best way to know how to do something could be to look around.

Reviewer: ```suggestion
We prefer consistency above all. So the best way to know how to do something could be to look around.
```

> ## Add new component

Reviewer: ```suggestion
## Adding a new component
```

Before writing a component you must know what problem it is supposed to solve and in what context to use it.

> Before add new component, please see if the component does not already exist, or the problem is not already solved by another component.

Reviewer: ```suggestion
Before adding a new component, please see if the component does not already exist, or if the problem is not already solved by another component.
```

- Create a branch (`git checkout -b branch_name`)
- Create a new file related to your component `src/components/ComponentName/ComponentName.tsx`
    - First of all define all properties that your component need to work, then type them the most strongly possible
    - Add comment on all your properties (those comments will be displayed in Storybook)
    - Write related logic of the component
    - Add style in your component with [styled components](https://styled-components.com/docs).

>     - Add if necessary icons in the folder `src/icons/` (height and weight should be configurable)

Reviewer: ```suggestion
    - Add (if necessary) icons in the folder `src/icons/` (height and weight should be configurable)
```

Reviewer 2: color should also be configurable

>     - Add forwardRef management on your component (https://fr.reactjs.org/docs/forwarding-refs.html)

Reviewer 3: ```suggestion
    - Add forwardRef management on your component (https://reactjs.org/docs/forwarding-refs.html)
```

        - Use `control` when your property expects a value (string, number, boolean, enum ...), you can retrieve full list of control types here: https://storybook.js.org/docs/react/essentials/controls#annotation.
        - Use `action` when your property expects a callback relative to an action on the component
    - Define `arg` when you want a specific default value
    - Write all variations of your component
- Create a new file src/components/`ComponentName/ComponentName.unit.tsx`
    - Add unit tests to validate all component behaviours
- Commit and push your change (Github Action will be automatically launched)
- Github Action will assign you a pull request to alert you that new stories found, review it and merge it when it looks good for you
- Github Action will deploy a new version of the storybook give the url to designer for review

## Update style of component

If you update the style of one component, the visual test will fail. Github Action will assign you a pull request to alert you that component changed, review it and merge it when it looks good for you.

Github Action will deploy a new version of the storybook give the url to designer for review.

> ## Component guidelines

Reviewer: ```suggestion
## Components guidelines
```
I'd put either all occurrences plural
